cylc/flow/task_remote_mgr.py

```diff
@@ -300,44 +301,93 @@ def construct_remote_tidy_ssh_cmd(
         cmd = construct_ssh_cmd(cmd, platform, host, timeout='10s')
         return cmd, host
 
+    @staticmethod
+    def _get_remote_tidy_targets(
+        platform_names: List[str],
+        install_targets: Set[str]
+    ) -> Dict[str, List[Dict[str, Any]]]:
+        """Finds valid platforms for install targets, warns about in invalid
+        install targets.
+
+        logs:
+            A list of install targets where no platform can be found.
+
+        returns:
+            A mapping of install targets to valid platforms only where
+            platforms are available.
+        """
+        if not platform_names and install_targets:
+            install_targets_map: Dict[str, List[Dict[str, Any]]] = {
+                t: [] for t in install_targets}
+        else:
+            install_targets_map = get_install_target_to_platforms_map(
+                platform_names, quiet=True)
+
+        # If we couldn't find a platform for a target, we cannot tidy it -
+        # raise an Error:
+        unreachable_targets = [
+            t for t, v in install_targets_map.items()
+            if not v
+        ] + list(install_targets.difference(install_targets_map))
```

Comment:
This seems more complicated than it need be
```suggestion
            install_targets_map: Dict[str, List[Dict[str, Any]]] = {
                t: [] for t in install_targets
            }
            unreachable_targets = install_targets
        else:
            install_targets_map = get_install_target_to_platforms_map(
                platform_names, quiet=True)
            unreachable_targets = install_targets.union(
                t for t, v in install_targets_map.items()
                if not v
            )
```
However I'm not sure the `unreachable_targets` in the second branch is correct - can I talk to you about this?

Reply:
sure, not sure how much I can remember of this...

Reply:
In words - it's an unreachable target if:
- it's in the install targets list by not the install targets map
- it's in the install targets map and has no platforms.